src/Server/Provision/ServerProvisioner.php

```diff
@@ -163,26 +163,35 @@ public function check()
      */
     private function checkEntities()
     {
-        if ($entities = $this->getEntities()) {
-            return $entities;
+        $isCheck = true;
+        if ($entities = $this->getEntities($isCheck)) {
+            return $entities[0];
         }
 
         throw new \Exception('No matching Entity found in inventory');
     }
 
     /**
-     * @return Entity|void
+     * @param boolean $isCheck
+     *
+     * @return array|void
      */
-    public function getEntities()
+    public function getEntities($isCheck = false)
```

Comment:
1. isCheck doesn't make much sense as a variable name.
2. [boolean arguments are bad](https://softwareengineering.stackexchange.com/a/148058/214792), it'd be better to have two separate functions and have a third (private) function with the logic that's shared between them.
3. Is this ever used? if not we shouldn't add it. If so, we need to split this function into two and name them better as mentioned above.